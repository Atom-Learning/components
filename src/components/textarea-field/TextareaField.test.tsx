import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { TextareaField } from '.'

describe('TextareaField component', () => {
  it('renders', async () => {
    const { container } = render(
      <TextareaField name="description" id="description" />
    )

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <TextareaField name="description" id="description" />
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders with an error', async () => {
    const { container } = render(
      <TextareaField
        name="description"
        id="description"
        error="Error Message"
      />
    )

    expect(container).toMatchSnapshot()
  })
})
