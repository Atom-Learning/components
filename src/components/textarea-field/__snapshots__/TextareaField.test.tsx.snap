// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaField component renders 1`] = `
.sx2cs53 {
  display: flex;
}

.sx2cs53-qom1k {
  justify-content: space-between;
  align-items: center;
  margin-bottom: var(--sx-space-3);
}

.sxbagj1 {
  color: var(--sx-colors-tonal500);
  display: block;
  font-family: var(--sx-fonts-body);
  font-weight: 600;
  margin: 0;
}

.sxbagj18bitc--size-md {
  font-size: var(--sx-fontSizes-md);
  line-height: 1.5;
}

.sxbagj18bitc--size-md::before {
  content: '';
  margin-bottom: -0.3864em;
  display: table;
}

.sxbagj18bitc--size-md::after {
  content: '';
  margin-top: -0.3864em;
  display: table;
}

.sxw9jbw {
  box-shadow: none;
  font-size: var(--sx-fontSizes-md);
  -webkit-appearance: none;
  appearance: none;
  border: 1px solid var(--sx-colors-tonal400);
  border-radius: var(--sx-radii-0);
  box-sizing: border-box;
  color: var(--sx-colors-tonal600);
  font-family: var(--sx-fonts-body);
  font-weight: 400;
  line-height: 1.4;
  min-height: var(--sx-sizes-7);
  padding-left: var(--sx-space-3);
  padding-right: var(--sx-space-3);
  padding-top: calc(var(--sx-space-2) + var(--sx-space-1));
  padding-bottom: calc(var(--sx-space-2) + var(--sx-space-1));
  resize: vertical;
  transition: all 75ms ease-out;
  width: 100%;
}

.sxw9jbw:focus {
  border-color: var(--sx-colors-primary);
  outline: none;
}

.sxw9jbw[disabled] {
  background-color: var(--sx-colors-tonal100);
  color: var(--sx-colors-tonal400);
  cursor: not-allowed;
}

.sxw9jbw::placeholder {
  color: var(--sx-colors-tonal300);
  opacity: 1;
}

<div>
  <form
    aria-label="form"
    class="sx03kze"
  >
    <div
      class="sx03kze"
    >
      <div
        class="sx2cs53 sx2cs53-qom1k"
      >
        <label
          class="sxbagj1 sxbagj18bitc--size-md"
          for="description"
        />
      </div>
      <textarea
        class="sxw9jbw"
        id="description"
        name="description"
        placeholder="placeholder"
      />
    </div>
  </form>
</div>
`;

exports[`TextareaField component renders with an error 1`] = `
.sx2cs53 {
  display: flex;
}

.sx2cs53-qom1k {
  justify-content: space-between;
  align-items: center;
  margin-bottom: var(--sx-space-3);
}

.sxbagj1 {
  color: var(--sx-colors-tonal500);
  display: block;
  font-family: var(--sx-fonts-body);
  font-weight: 600;
  margin: 0;
}

.sxbagj18bitc--size-md {
  font-size: var(--sx-fontSizes-md);
  line-height: 1.5;
}

.sxbagj18bitc--size-md::before {
  content: '';
  margin-bottom: -0.3864em;
  display: table;
}

.sxbagj18bitc--size-md::after {
  content: '';
  margin-top: -0.3864em;
  display: table;
}

.sxw9jbw {
  box-shadow: none;
  font-size: var(--sx-fontSizes-md);
  -webkit-appearance: none;
  appearance: none;
  border: 1px solid var(--sx-colors-tonal400);
  border-radius: var(--sx-radii-0);
  box-sizing: border-box;
  color: var(--sx-colors-tonal600);
  font-family: var(--sx-fonts-body);
  font-weight: 400;
  line-height: 1.4;
  min-height: var(--sx-sizes-7);
  padding-left: var(--sx-space-3);
  padding-right: var(--sx-space-3);
  padding-top: calc(var(--sx-space-2) + var(--sx-space-1));
  padding-bottom: calc(var(--sx-space-2) + var(--sx-space-1));
  resize: vertical;
  transition: all 75ms ease-out;
  width: 100%;
}

.sxw9jbw:focus {
  border-color: var(--sx-colors-primary);
  outline: none;
}

.sxw9jbw[disabled] {
  background-color: var(--sx-colors-tonal100);
  color: var(--sx-colors-tonal400);
  cursor: not-allowed;
}

.sxw9jbw::placeholder {
  color: var(--sx-colors-tonal300);
  opacity: 1;
}

<div>
  <form
    aria-label="form"
    class="sx03kze"
  >
    <div
      class="sx03kze"
    >
      <div
        class="sx2cs53 sx2cs53-qom1k"
      >
        <label
          class="sxbagj1 sxbagj18bitc--size-md"
          for="description"
        />
      </div>
      <textarea
        class="sxw9jbw"
        error="Error Message"
        id="description"
        name="description"
        placeholder="placeholder"
      />
    </div>
  </form>
</div>
`;
